components/clubhouse_io/actions/create-story/create-story.js

```diff
@@ -0,0 +1,371 @@
+
+module.exports = {
+  key: "clubhouse-create-story",
+  name: "Create Story",
+  description: "Creates a new story in your clubhouse.",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    clubhouse,
+    archived: {
+      type: "boolean",
+      label: "Archived",
+      description: "Controls the story’s archived state.",
+      optional: true,
+    },
+    comments: {
+      type: "string",
+      label: "Comments",
+      description:
+        "An array with comment objects to add to the story. Each comment object must have the [CreateStoryCommentParams](https://clubhouse.io/api/rest/v3/#CreateStoryCommentParams) structure. Alternatively, provide a string that will `JSON.parse` to an array of comment objects. ",
+      optional: true,
+    },
+    completedAtOverride: {
+      type: "string",
+      label: "Completed at Override Date",
+      description:
+        "A manual override for the time/date the Story was completed.",
+      optional: true,
+    },
+    createdAt: {
+      type: "string",
+      label: "Created at Date",
+      description: "The time/date the Story was created.",
+      optional: true,
+    },
+    dueDate: {
+      type: "string",
+      label: "Due Date",
+      description: "The due date of the story.",
+      optional: true,
+    },
+    description: {
+      type: "string",
+      label: "Description",
+      description: "The description of the story.",
+      optional: true,
+    },
+    epicId: {
+      type: "integer",
+      label: "Epic ID",
+      description: "The unique identifier of the epic the story belongs to.",
+      async options() {
+        const options = [];
+        const epics = await this.clubhouse.listEpics();
+        epics.forEach((epic) => {
+          options.push({
+            label: epic.name,
+            value: epic.id,
+          });
+        });
+        return options;
+      },
+      optional: true,
+    },
+    estimate: {
+      type: "integer",
+      label: "Estimate",
+      description:
+        "The numeric point estimate of the story. Can also be null, which means unestimated.",
+      optional: true,
+    },
+    externalId: {
+      type: "integer",
+      label: "External Id",
+      description:
+        "This field can be set to another unique ID. In the case that the Story has been imported from another tool, the ID in the other tool can be indicated here.",
+      optional: true,
+    },
+    externalLinks: {
+      type: "string[]",
+      label: "External Links",
+      description: "A string array of External Links associated with this story.",
+      optional: true,
+    },
+    fileIds: {
+      type: "integer[]",
+      label: "File Ids",
+      description: "An integer array of IDs of files attached to the story.",
+      optional: true,
+    },
+    followerIds: {
+      type: "string[]",
+      label: "Follower Ids",
+      description: "A string array of UUIDs of the followers of this story.",
+      optional: true,
```

Comment:
sorry @compwright but how would you add options dynamically to a prop that is a string?
I can see that working for props that request only one options from the many populated dynamic, but in this case for "Follower Ids" I don't see it feasable.
Same for "Linked File Ids" (which I mistaknly added thinking it was singuar ID instead of Ids"

Reply:
@sergioeliot2039 I was under the impression that this would work exactly the same as it would for a normal non-array field. @dylburger can you confirm?

Reply:
Yes async options are supported for both strings and arrays of strings. If you use async options but set the type to `string[]`, the user can select multiple options from the list.

Reply:
I've been wrong all this time!
Ok I'll use the dynamic options

Reply:
I refactored the `async options` to a method in the appfile instead, otherwise I would have to override type, label, description, and name each prop differently.